### PR TITLE
POC-846 Add signed message in loan creation

### PR DESCRIPTION
**Pull request by cfcfs**
## Purpose of this PR 🎯

-   [ ] Feature;
-   [ ] Bugfix;
-   [ ] Tests;
-   [x ] Refactoring;
-   [ ] Build or CI/CD; 
-   [ ] Documentation;
-   [ ] Code Styling;
-   [ ] Other. Please describe:

## Changes 📝

Add signed message support for loan creation (`reserve`)
Remove the `validate`, `invalidate` and `cancelPendingLoan` functions from Loans
Remove the `LoanValidated` and `LoanInvalidated` events

## Test Coverage 🧻

Adapted tests to new function signature

## Does this PR introduce a breaking change? ⚠️

-   [ ] No
-   [ x] Yes

New version of Loans contract
## Reviewers 🦺

@DioPires

### CODE REVIEW

File: contracts/Loans.vy [hunk 1]
```diff
@@ -684,6 +711,24 @@ def reserve(
...
+                RESERVE_TYPE_HASH,
+                _amount,
+                _interest,
+                _maturity,
+                keccak256(slice(_abi_encode(collaterals_data_hash), 32*2, 32*len(_collaterals))),
+                _deadline,
+                ))
+
+    sig_hash: bytes32 = keccak256(concat(convert("\x19\x01", Bytes[2]), _abi_encode(self.reserve_sig_domain_separator, data_hash)))
+    signer: address = ecrecover(sig_hash, _v, _r, _s)
```
Comment (DioPires) [inline contracts/Loans.vy:729]: Can we put this in a `_recoverSigner` method to keep the `reserve` method simpler?

**Review by DioPires — CHANGES_REQUESTED** [review]
We can delete the `PendingLoanCanceled` event in `Loans.vy`

File: contracts/Loans.vy (hunk 1 above)
Comment (cfcfs) [reply contracts/Loans.vy]: Done

**Review by DioPires — APPROVED** [review]
The tests are failing because of the issue related to NFTX/Sushi integration tests. Everything else LGTM 👍

**Pull request merged**